research/python/src/data/provider.py

```
@@ -0,0 +1,1077 @@
+            Dict containing market state, trading hours, etc.
+        """
+        cache_key = "market_status"
+        cached = self._get_cached_data(cache_key)
+        if cached:
+            return cached
+
+        try:
+            status = self._retry_operation(self.nse.market_status)
+            self._cache_data(cache_key, status)
+            return status
+        except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)